### PR TITLE
SAK-29160 add quota & storage usage to content feed

### PR DESCRIPTION
**Pull request by bkirschn**

### CODE REVIEW

**Comment by buckett** [comment]
Should quota be made available to non-maintainers of the site?

**Comment by bkirschn** [comment]
I'm not sure why we hide quote from non-maintainers of the site, but  I changed the feed to be consistent with the UI, so non-site-maintainers no longer can see the quota

**Comment by jonespm** [comment]
Possibly because it's both a slight performance hit to calculate the storage used when it's not necessary (on large resource collections) and someone might abuse it if they saw it was no quota. Just a guess though. Good for the consistency at least.

File: content/content-tool/tool/src/java/org/sakaiproject/content/entityproviders/ContentEntityProvider.java [hunk 1]
```diff
@@ -179,6 +181,10 @@ public boolean entityExists(String id) {
 						ContentItem item = new ContentItem();
 						item.setType("collection");
 						item.setSize(contentHostingService.getCollectionSize(id));
+						if (allowUpdateSite) // to be consistent with UI
+							item.setQuota(Long.toString(contentHostingService.getQuota(collection)));
+						item.setUsage(Long.toString(collection.getBodySizeK() * 1024));
```
Comment (buckett) [inline content/content-tool/tool/src/java/org/sakaiproject/content/entityproviders/ContentEntityProvider.java]: The quota usage looks like it should always be fast as the columns that pre-calculate it has to exist.

`org.sakaiproject.content.impl.BaseContentService.BaseCollectionEdit#getBodySizeK()`

Comment (jonespm) [inline content/content-tool/tool/src/java/org/sakaiproject/content/entityproviders/ContentEntityProvider.java]: Yeah, the columns in the database do exist, but the call to getBodySizeK has to run a select/aggregate function (SUM) on all of the contents in that collection to get the size, and I didn't see anywhere that this sum is cached.

select SUM(FILE_SIZE) from CONTENT_RESOURCE where CONTEXT = ?

For the regular case, probably not too bad to run that, but I've seen huge content collections. ;)

Comment (ern) [inline content/content-tool/tool/src/java/org/sakaiproject/content/entityproviders/ContentEntityProvider.java]: FYI this is fairly easy to detect if a query is taking a while to run, plus the database optimizes this.

**Comment by ern** [comment]
Beth you need to update this PR as it can't be merged.

**Comment by bkirschn** [comment]
I'm gong to close this PR and re-open another -- my attempts to update this branch so it can be merged brought in way too much

**Pull request closed without merging**